command/cp.go

```diff
@@ -530,6 +530,10 @@ func (c Copy) Run(ctx context.Context) error {
 			if c.metadataDirective == "" {
 				// default to COPY
 				c.metadataDirective = "COPY"
+				if c.src.IsRemote() && c.dst.IsRemote() && c.contentType != "" {
+					// default to REPLACE for content type change
+					c.metadataDirective = "REPLACE"
+				}
```

Comment:
Does the `aws` cli shows the same behaviour ? I believe it defaults to `COPY` in all cases, the user explicitly needs to pass `REPLACE`: https://arc.net/l/quote/grqlrdlq

Reply:
# AWS CLI Test

## Object Metadata Before Copy
```shell
aws --endpoint-url=http://localhost:4566 s3api head-object --bucket copy-test --key dest/test.txt
```

```json
{
    "AcceptRanges": "bytes",
    "LastModified": "2024-07-22T08:46:26+00:00",
    "ContentLength": 12,
    "ETag": "\"90a6f0c908b76d95b8d9c0e6405b2fd5\"",
    "ContentType": "text/plain",
    "ServerSideEncryption": "AES256",
    "Metadata": {}
}
```

## Copy Without metadirective flag
```
aws --endpoint-url=http://localhost:4566 s3 cp  s3://copy-test/dest/test.txt s3://copy-test/dest/test.txt --content-type video/mp4
```

```
copy: s3://copy-test/dest/test.txt to s3://copy-test/dest/test.txt
```
## Object Metadata After Copy 
```
aws --endpoint-url=http://localhost:4566 s3api head-object --bucket copy-test --key dest/test.txt
```

```json
{
    "AcceptRanges": "bytes",
    "LastModified": "2024-07-22T08:46:40+00:00",
    "ContentLength": 12,
    "ETag": "\"90a6f0c908b76d95b8d9c0e6405b2fd5\"",
    "ContentType": "video/mp4",
    "ServerSideEncryption": "AES256",
    "Metadata": {}
}
```

Reply:
Thanks for the demonstration, the document agrees with your result: https://arc.net/l/quote/ksyjgxlm